Documentation/devel/distribution.md

```diff
@@ -0,0 +1,147 @@
+
+**NOTE:** I'm not sure if distribution is the best name for this concept or it should be called with other names like "image reference". So I'd like to hear your thoughts.
+
+A distribution represents the method to fetch an image starting from an input string. It doesn't mandates the image types. Usually a distribution can only provide one image type but some can provide different image types. For example:
+
+* Some docker registries also provides OCI images.
+* Rkt can internally implement the docker distribution to natively fetch a docker/OCI image from a registry or an ACI converted by docker2aci.
+
+Before this concept was introduced, rkt used the ImageType concept, the ImageType was mapped to the input image string format and internally covered multiple concepts like distribution, transport and image type (hidden since all are now appc ACIs).
+
+The distribution concept is used as the primary information in the different layers of rkt (fetching but also for references in a CAS/ref store).
+
+## Distribution types
+
+The distribution can be logically split in two types:
+
+* Name based distribution
+* Location based distribution
```

Comment:
I guess a better categorization here is indirect/direct. Name-based can perform additional indirect steps (eg. discovery, assembling, signature fetching) while location-based seems to suit direct go-to scenarios.